boot/util.ml

```diff
@@ -59,8 +59,22 @@ let canonical_path_name p =
     (* We give up to find a canonical name and just simplify it... *)
     Filename.concat current p
 
+let find_in_PATH f =
+  match Sys.getenv_opt "PATH" with
+  | None -> None
+  | Some paths ->
+    let sep = if Coq_config.arch_is_win32 then ';' else ':' in
+    let paths = String.split_on_char sep paths in
+    paths |> List.find_opt (fun path ->
+        Sys.file_exists (if path = "" then f else Filename.concat path f))
+
 let coqbin =
-  canonical_path_name (Filename.dirname Sys.executable_name)
+  (* avoid following symlinks if possible (Sys.executable_name followed symlinks) *)
```

Comment:
not sure if this is the right way to work
maybe we should be following symlinks and special case when they point in the dune build layout?